src/test/java/bio/terra/service/search/SearchTest.java

```diff
@@ -0,0 +1,105 @@
+package bio.terra.service.search;
+
+import bio.terra.common.category.Unit;
+import bio.terra.model.SearchIndexRequest;
+import bio.terra.service.resourcemanagement.google.GoogleProjectResource;
+import bio.terra.service.snapshot.Snapshot;
+import bio.terra.service.snapshot.SnapshotTable;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.mockito.Mockito.when;
+
+@Category(Unit.class)
+public class SearchTest {
+    private static final String sqlQuery = "SELECT GENERATE_UUID() uuid, CURRENT_TIMESTAMP() as now FROM UNNEST" +
+        "(GENERATE_ARRAY(1, 3));";
+
+    @Mock
+    private BigQueryPdao bigQueryPdao;
+
+    @Mock
+    private RestHighLevelClient client;
+
+    private SearchService service;
+
+    private SearchIndexRequest searchIndexRequest;
+    private Snapshot snapshot;
+    private List<Map<String, Object>> values;
+
+    @Before
+    public void setup() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        service = new SearchService(bigQueryPdao, client);
+
+        searchIndexRequest = getSearchIndexRequest();
+        snapshot = getSnapshot();
+        values = getSnapshotTableData();
+    }
+
+    @Test
+    public void indexSnapshotTest() throws Exception {
+        mockGetSnapshotTableData();
+        //service.indexSnapshot(snapshot, searchIndexRequest);
+    }
+
+    private void mockGetSnapshotTableData() throws InterruptedException {
+        when(bigQueryPdao.getSnapshotTableData(snapshot, searchIndexRequest.getSql()))
```

Comment:
Looking at the code, we will probably want a unit test for `BigQueryPdao.getSnapshotTableData()` too

Reply:
👍